cfme/tests/test_appliance.py

```diff
@@ -125,22 +152,20 @@ def test_iptables_rules(appliance, proto, port):
         casecomponent: Appliance
         upstream: no
     """
-    # get the current iptables state, nicely formatted for us by iptables-save
-    result = appliance.ssh_client.run_command('iptables-save')
-    # get everything from the input chain
-    input_rules = filter(lambda line: line.startswith('-A IN'), result.output.splitlines())
-
-    # filter to make sure we have a rule that matches the given proto and port
-    def rule_filter(rule):
-        # iptables-save should put all of these in order for us
-        # if not, this can be broken up into its individual components
-        matches = [
-            '-p {proto}',
-            '-m {proto} --dport {port}',
-            '-j ACCEPT'
-        ]
-        return all([match.format(proto=proto, port=port) in rule for match in matches])
-    assert filter(rule_filter, input_rules)
+    manageiq_zone = "manageiq"
+
+    def get_def_iface_zone():
+        default_iface_zone_cmd = appliance.ssh_client.run_command(
+            "firewall-cmd --get-zone-of-interface {}".format(default_iface)
+        )
+        assert default_iface_zone_cmd.success
+        return default_iface_zone_cmd.output.strip()
+
+    assert get_def_iface_zone() == manageiq_zone
+
+    assert appliance.ssh_client.run_command("systemctl restart firewalld").success
```

Comment:
Please add a systemd object to appliance, and interact with firewalld through it. We have a Systemd unit class to handle this in a more consistent/programmatic way - You can directly call restart and wait for it to be running again after.

Reply:
@jaryn I see you added the service object, but aren't using it here. Miss a commit?